### PR TITLE
kv: fix err check

### PR DESCRIPTION
**Pull request by Tjianke**
### What problem does this PR solve?

Issue Number: #22979  

Problem Summary:

### What is changed and how it works?

What's Changed: 
- added basic err check
- some assert failed so I commented them with `// to fix:`. Needs further action, either ignore these error or fix tests.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- No release note

### CODE REVIEW

File: kv/mock_test.go [hunk 1]
```diff
@@ -56,7 +56,8 @@ func (s testMockSuite) TestInterface(c *C) {
 		_, err = transaction.IterReverse(Key("lock"))
 		c.Check(err, IsNil)
 	}
-	transaction.Commit(context.Background())
+	// to fix: cannot assert err, e.g. KV error safe to retry %s
+	_ = transaction.Commit(context.Background())
```
Comment (zimulala) [inline kv/mock_test.go:60]: PTAL @tiancaiamao

**Comment by tiancaiamao** [comment]
/LGTM

Do we do this to enable the lint check for the test code?

**Comment by Tjianke** [comment]
> /LGTM
> 
> Do we do this to enable the lint check for the test code?

Yes.

**Comment by tiancaiamao** [comment]
PTAL @zimulala

File: store/tikv/unionstore/memdb_norace_test.go [hunk 2]
```diff
@@ -50,11 +51,13 @@ func (s testMemDBSuite) TestRandom(c *C) {
 		op := rand.Float64()
 		if op < 0.35 {
 			p1.DeleteKey(k)
-			p2.Delete(k)
+			// to fix: delete key not found
+			_ = p2.Delete(k)
```
Comment (zimulala) [inline store/tikv/unionstore/memdb_norace_test.go:55]: @tiancaiamao 
I think we need to check this error.

File: kv/mock_test.go (hunk 1 above)
Comment (tiancaiamao) [reply kv/mock_test.go:61]: This code just check the API interface, even if here returns an error, it doesn't matter.

File: store/tikv/unionstore/memdb_norace_test.go (hunk 2 above)
Comment (xhebox) [reply store/tikv/unionstore/memdb_norace_test.go:55]: I think it does not need to be so strict, there is already `_ = p2.Put(k, k)` in the previous code. And it is a test case.

**Comment by tiancaiamao** [comment]
/LGTM

Please resolve the conflicts

**Comment by tisonkun** [comment]
After integrate with current master, it seems `store/tikv` files move to `client-go`, and `mock_test.go` already changed.

Closed.

**Pull request closed without merging**